pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/BaseContext.java

```diff
@@ -194,11 +195,27 @@
     void recordMetric(String metricName, double value);
 
     /**
-     * Get the pulsar client.
+     * Get the pre-configured pulsar client.
+     *
+     * You can use this client to access Pulsar cluster.
+     * The Function will be responsible for disposing this client.
      *
      * @return the instance of pulsar client
      */
     default PulsarClient getPulsarClient() {
         throw new UnsupportedOperationException("not implemented");
     }
+
+    /**
+     * Get the pre-configured pulsar client builder.
+     *
+     * You can use this Builder to setup client to connect to the Pulsar cluster.
+     * The Function will be responsible for disposing any PulsarClient created by this builder.
+     *
+     * @return the instance of pulsar client builder.
+     */
+    default ClientBuilder getPulsarClientBuilder() {
```

Comment:
@nlu90 You need to implement this method in the sub-classes.

Reply:
working on it